functions/go/my-function/myfunction/walk.go

```diff
@@ -0,0 +1,69 @@
+package myfunction
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// visitor is implemented by structs which need to walk the configuration.
```

Comment:
I guess most functions won't need to use visitor pattern, since most functions are for special purpose (i.e. only doing one thing). I expect general purpose functions like `apply-setters` would be much fewer.

Also IMO this visitor interface should be part of kyaml.